shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,951 @@
+        elif len(pickings) > 1:
+            return self._response_for_select_document(
+                pickings=pickings,
+                message=self.msg_store.multiple_picks_found_select_manually(),
+            )
+        # If no available picking with the right state has been found,
+        # return an error
+        return self._response_for_select_document(
+            message=self.msg_store.product_not_found_in_pickings()
+        )
+
+    def _select_document_from_packaging(self, packaging):
+        """Select the document by packaging
+
+        next states:
+            - set_lot: a single picking has been found for this packaging
+            - select_document: A single or no pickings has been found for this packaging
+        """
+        move_lines = self._move_line_by_packaging(packaging)
+        pickings = move_lines.mapped("move_id.picking_id")
+        if len(pickings) == 1:
+            pickings.user_id = self.env.user
+            if packaging.product_id.tracking not in ("lot", "serial"):
+                return self._response_for_set_quantity(pickings, move_lines)
+            return self._response_for_set_lot(pickings, move_lines)
+        elif len(pickings) > 1:
+            return self._response_for_select_document(
+                pickings=pickings,
+                message=self.msg_store.multiple_picks_found_select_manually(),
+            )
+        # If no available picking with the right state has been found,
+        # return a barcode not found error message
+        return self._response_for_select_document(
+            message=self.msg_store.no_transfer_for_packaging(),
+        )
+
+    def _select_line_from_product(self, picking, product):
+        lines = picking.move_line_ids.filtered(lambda l: l.product_id == product)
+        if not lines:
+            return self._response_for_select_line(
+                picking,
+                message=self.msg_store.product_not_found_in_picking(),
+            )
+        picking.user_id = self.env.user
+        if product.tracking not in ("lot", "serial"):
+            return self._response_for_set_quantity(picking, lines)
+        return self._response_for_set_lot(picking, lines)
+
+    def _select_line_from_packaging(self, picking, packaging):
+        lines = picking.move_line_ids.filtered(
+            lambda l: l.package_id.product_packaging_id == packaging
+        )
+        if not lines:
+            return self._response_for_select_line(
+                picking,
+                message=self.msg_store.packaging_not_found_in_picking(),
+            )
+        picking.user_id = self.env.user
+        if packaging.product_id.tracking not in ("lot", "serial"):
+            return self._response_for_set_quantity(picking, lines)
+        return self._response_for_set_lot(picking, lines)
+
+    def _order_stock_picking(self):
+        return "scheduled_date ASC"
+
+    def _scan_document__scan_picking(self, barcode):
```

Comment:
maybe replace "scan" w/ "by" in all these methods?
```suggestion
    def _scan_document__by_picking(self, barcode):
```